src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -329,6 +329,34 @@ bool ClusterResourceScheduler::GetNodeResources(int64_t node_id,
 
 int64_t ClusterResourceScheduler::NumNodes() { return nodes_.size(); }
 
+void ClusterResourceScheduler::AddLocalResource(
+                                                      const std::string &resource_name,
+                                                      double resource_total) {
+  string_to_int_map_.Insert(resource_name);
+  int64_t resource_id = string_to_int_map_.Get(resource_name);
+
+  if (local_resources_.custom_resources.find(resource_id) != local_resources_.custom_resources.end()) {
+    FixedPoint total(resource_total);
+    auto &instances = local_resources_.custom_resources[resource_id];
+    instances.total[0] += total;
```

Comment:
Can you comment what this index 0 mean?

Reply:
We always only use index 0 for non-unit resources (we do this in a lot of places)

```
src/ray/raylet/scheduling/cluster_task_manager_test.cc:459:            (expected_load[0] == load.num_infeasible_requests_queued() &&
src/ray/raylet/scheduling/cluster_task_manager_test.cc:467:        RAY_LOG(ERROR) << expected_load[0] << "\t" <<
src/ray/raylet/scheduling/cluster_resource_data.cc:72:    task_request.predefined_resources[0].demand = 0;
src/ray/raylet/scheduling/cluster_resource_data.cc:73:    task_request.predefined_resources[0].soft = false;
src/ray/raylet/scheduling/cluster_resource_scheduler.cc:443:    instance_list->total[0] = instance_list->available[0] = total;
src/ray/raylet/scheduling/cluster_resource_scheduler.cc:510:    if (available[0] >= remaining_demand) {
src/ray/raylet/scheduling/cluster_resource_scheduler.cc:511:      available[0] -= remaining_demand;
src/ray/raylet/scheduling/cluster_resource_scheduler.cc:512:      (*allocation)[0] = remaining_demand;
src/ray/raylet/scheduling/cluster_resource_scheduler.cc:516:        available[0] = 0;
```